pkg/customresourcestate/registry_factory.go

```diff
@@ -217,7 +217,35 @@ func (c *compiledGauge) Values(v interface{}) (result []eachValue, errs []error)
 	switch iter := v.(type) {
 	case map[string]interface{}:
 		for key, it := range iter {
-			ev, err := c.value(it)
+			// Try to deduce `valueFrom`'s value from the current element.
+			var ev *eachValue
+			var err error
+			var didResolveValueFrom bool
+			sValueFrom := c.ValueFrom.String()
+			if !strings.Contains(sValueFrom, ",") &&
+				sValueFrom[0] == '[' &&
+				len(sValueFrom) > 2 {
+				extractedValueFrom := sValueFrom[1 : len(sValueFrom)-1]
+				if key == extractedValueFrom {
+					gotFloat, err := toFloat64(it, c.NilIsZero)
+					if err != nil {
+						onError(fmt.Errorf("[%s]: %w", key, err))
+						continue
+					}
+					labels := make(map[string]string)
+					ev = &eachValue{
+						Labels: labels,
+						Value:  gotFloat,
+					}
+					didResolveValueFrom = true
+				}
+			}
+			if !didResolveValueFrom {
+				ev, err = c.value(it)
```

Comment:
do we need to handle the error here as well?

Reply:
The error will get appended [here](https://github.com/kubernetes/kube-state-metrics/pull/1958/files#diff-1cbeb50a6b171ac66f4fad40018421010ed1733ea97fcb40b65a7ebd5e7e3d77R250).